### PR TITLE
iOS deleteAll() functionality

### PR DESCRIPTION
**Pull request by markrickert**
This PR adds functionality for deleting all keychain data associated with your app. This functionality is iOS only for now (anyone want to tackle this on Android?)

Also adds a button to the demo app showing usage and makes the demo app reference the node module from the cloned repo. instead of `^3.0.0`

### CODE REVIEW

**Comment by oblador** [comment]
I'm a bit hesitant to merge this as other third party libraries may be using the keychain for unrelated things like analytics or authentication. Can you think of a way of preventing that?

**Comment by markrickert** [comment]
I totally understand the hesitation. Maybe just a big red warning in the docs or something?  I mostly use this in `__DEV__` mode to reset my app state.

...the person who makes the hammer isn't responsible if someone hits their thumb with said hammer 🙃

File: KeychainExample/App.js [hunk 1]
```diff
@@ -69,6 +69,19 @@ export default class KeychainExample extends Component {
     }
   }
 
+  async deleteAll() {
```
Comment (koyta) [inline KeychainExample/App.js:72]: `deleteAll = () => {...}` and call it without creating an arrow function in render body `onPress={deleteAll}`

**Comment by oblador** [comment]
This should now be possible to achieve using from [`getAllGenericPasswordServices()`](https://github.com/oblador/react-native-keychain#getallgenericpasswordservices) from https://github.com/oblador/react-native-keychain/pull/396

**Pull request closed without merging**